database/src/main/scala/cromwell/database/slick/MetadataComponent.scala

```diff
@@ -0,0 +1,62 @@
+package cromwell.database.slick
+
+
+import java.sql.Timestamp
+
+import cromwell.database.obj.Metadatum
+
+trait MetadataComponent {
+  this: DriverComponent with WorkflowExecutionComponent =>
+
+  import driver.api._
+
+  class Metadata(tag: Tag) extends Table[Metadatum](tag, "METADATA_JOURNAL") {
+    def metadataId = column[Int]("METADATA_JOURNAL_ID", O.PrimaryKey, O.AutoInc)
+    def workflowExecutionUuid = column[String]("WORKFLOW_EXECUTION_UUID")
+    def key = column[String]("METADATA_KEY")
+    def callFqn = column[Option[String]]("METADATA_CALL_FQN")
+    def index = column[Option[Int]]("METADATA_CALL_INDEX")
+    def attempt = column[Option[Int]]("METADATA_CALL_ATTEMPT")
+    def value = column[Option[String]]("METADATA_VALUE")
+    def timestamp = column[Timestamp]("METADATA_TIMESTAMP")
+
+    override def * = (workflowExecutionUuid, key, callFqn, index, attempt, value, timestamp, metadataId.?) <>
+      (Metadatum.tupled, Metadatum.unapply)
+  }
+
+  protected val metadata = TableQuery[Metadata]
+
+  val metadataByWorkflowUuid = Compiled(
+    (workflowExecutionUuid: Rep[String]) => for {
+      metadatum <- metadata
+      if metadatum.workflowExecutionUuid === workflowExecutionUuid
+    } yield metadatum)
+
+  val metadataByWorkflowUuidAndKey = Compiled(
+    (workflowExecutionUuid: Rep[String], key: Rep[String]) => for {
+      metadatum <- metadata
+      if metadatum.workflowExecutionUuid === workflowExecutionUuid
+      if metadatum.key === key
+    } yield metadatum)
```

Comment:
Since they'd eventually show up in totally different parts of the metadata JSON, I would guess the former is probably most right. 

But OTOH I suspect the actual number of cases where keys are the same at workflow vs call scope are pretty minimal